### PR TITLE
NDRCN-380-Update Navigation

### PR DESCRIPTION
**Pull request by simms11**

### CODE REVIEW

File: app/controllers/BulkFileUploadController.scala [hunk 1]
```diff
@@ -215,8 +216,8 @@ class BulkFileUploadController @Inject()(
...
   )
   private def getBulkEntryDetails(answers: Option[UserAnswers]): Call = answers.flatMap(_ .get(CustomsRegulationTypePage)) match {
-    case Some(CustomsRegulationType.UnionsCustomsCodeRegulation)  => routes.ArticleTypeController.onPageLoad(NormalMode)
-    case Some(CustomsRegulationType.UKCustomsCodeRegulation) => routes.UkRegulationTypeController.onPageLoad(NormalMode)
-    case _ => routes.UkRegulationTypeController.onPageLoad(NormalMode)
+      case Some(CustomsRegulationType.UnionsCustomsCodeRegulation)  => routes.ArticleTypeController.onPageLoad(NormalMode)
+      case Some(CustomsRegulationType.UKCustomsCodeRegulation) => routes.UkRegulationTypeController.onPageLoad(NormalMode)
+      case _ => routes.UkRegulationTypeController.onPageLoad(NormalMode)
   }
 }
```
Comment (lmugnai) [inline app/controllers/BulkFileUploadController.scala:223]:
```suggestion
}

```

File: app/navigation/Navigator.scala [hunk 2]
```diff
@@ -92,20 +92,23 @@ class Navigator @Inject()() {
       case  _  => routes.RepaymentTypeController.onPageLoad(NormalMode)
     }
 
-  private def getEntryDetails(answers: UserAnswers): Call = answers.get(CustomsRegulationTypePage) match {
-    case Some(CustomsRegulationType.UnionsCustomsCodeRegulation)  => {
-      answers.get(NumberOfEntriesTypePage).contains(NumberOfEntriesType.Single) match {
-        case true => routes.ArticleTypeController.onPageLoad (NormalMode)
-        case _=> routes.BulkFileUploadController.showFileUpload
-      }
+  private def getEntryDetails(answers: UserAnswers): Call =
```
Comment (lmugnai) [inline app/navigation/Navigator.scala:95]: Can we refactor this? Maybe just match on answers.get(CustomsRegulationTypePage) and check the CustomsRegulationType. 'contains' is good for checkboxes.

File: app/navigation/Navigator.scala [hunk 3]
```diff
@@ -92,20 +92,23 @@ class Navigator @Inject()() {
...
+      case true => routes.UkRegulationTypeController.onPageLoad(NormalMode)
+      case false => routes.ArticleTypeController.onPageLoad(NormalMode)
     }
-    case Some(CustomsRegulationType.UKCustomsCodeRegulation) => {
-      answers.get(NumberOfEntriesTypePage).contains(NumberOfEntriesType.Single) match {
-        case true => routes.UkRegulationTypeController.onPageLoad (NormalMode)
-        case _=> routes.BulkFileUploadController.showFileUpload
-      }
+
+  private def getArticleType(answers: UserAnswers) : Call =
```
Comment (lmugnai) [inline app/navigation/Navigator.scala:101]: Can we refactor this? Maybe just match on answers.get(NumberOfEntriesTypePage) and check the NumberOfEntriesType.

File: app/navigation/Navigator.scala [hunk 4]
```diff
@@ -92,20 +92,23 @@ class Navigator @Inject()() {
...
+    answers.get(NumberOfEntriesTypePage).contains(NumberOfEntriesType.Multiple) match {
+      case true => routes.BulkFileUploadController.showFileUpload ()
+      case false => routes.EntryDetailsController.onPageLoad (NormalMode)
+    }
+
+  private def getUKRegulationType(answers: UserAnswers) : Call =
```
Comment (lmugnai) [inline app/navigation/Navigator.scala:107]: Do we need this method if it's identical to getArticleType? Maybe simply rename the method name make sense for both scenarios?

File: test/navigation/NavigatorSpec.scala [hunk 5]
```diff
@@ -100,13 +100,13 @@ class NavigatorSpec extends SpecBase with ViewBehaviours {
 
       }
 
-      "go to BulkFileUpload page after the customsRegulationType page when the UnionsCustomsCodeRegulation has been selected" in {
+      "go to ArticleTypeController page after the customsRegulationType page when the UnionsCustomsCodeRegulation has been selected" in {
```
Comment (lmugnai) [inline test/navigation/NavigatorSpec.scala:103]: Do we have a test like this for the UkRegulationTypeController?

**Review by lmugnai — CHANGES_REQUESTED** [review]
Give me a shout @simms11 if you want to go over any of these comments.

File: app/navigation/Navigator.scala (hunk 2 above)
Comment (simms11) [reply app/navigation/Navigator.scala:95]: @lmugnai  is this comment specifically related to get getEntryDetails?
image

File: app/controllers/BulkFileUploadController.scala [hunk 6]
```diff
@@ -24,7 +24,7 @@ import connectors.{UpscanInitiateConnector, UpscanInitiateRequest}
 import controllers.actions._
 import models.FileType.Bulk
 import models.{CustomsRegulationType, FileVerificationStatus, NormalMode, S3UploadError, UpscanNotification, UserAnswers}
-import pages.CustomsRegulationTypePage
+import pages.{CustomsRegulationTypePage}
```
Comment (lmugnai) [inline app/controllers/BulkFileUploadController.scala:27]:
```suggestion
import pages.CustomsRegulationTypePage
```

File: app/navigation/Navigator.scala [hunk 7]
```diff
@@ -92,18 +92,18 @@ class Navigator @Inject()() {
...
-    case Some(CustomsRegulationType.UKCustomsCodeRegulation) => {
-      answers.get(NumberOfEntriesTypePage).contains(NumberOfEntriesType.Single) match {
-        case true => routes.UkRegulationTypeController.onPageLoad (NormalMode)
-        case _=> routes.BulkFileUploadController.showFileUpload
-      }
+
+  private def getReasonForRepayment(answers: UserAnswers) : Call = {
+    answers.get(NumberOfEntriesTypePage) match {
+      case Some (NumberOfEntriesType.Multiple) => routes.BulkFileUploadController.showFileUpload ()
+      case Some (NumberOfEntriesType.Single) => routes.EntryDetailsController.onPageLoad (NormalMode)
```
Comment (lmugnai) [inline app/navigation/Navigator.scala:105]:
```suggestion
      case Some (NumberOfEntriesType.Single) => routes.EntryDetailsController.onPageLoad(NormalMode)
```

**Review by lmugnai — CHANGES_REQUESTED** [review]
Hi @simms11 the backlink doesn't seem to be behaving correctly when going back from the entrydetails page. Seems to be stuck in a loop.

**Review by lmugnai — APPROVED** [review]

**Pull request merged**